src/azul/project/hca/transformers.py

```diff
@@ -437,6 +437,44 @@ def transform(self,
             yield self._contribution(bundle, contents, file.document_id)
 
 
+class CellSuspensionTransformer(Transformer):
+
+    def entity_type(self) -> str:
+        return 'cellsuspensions'
+
+    def transform(self,
+                  uuid: str,
+                  version: str,
+                  manifest: List[JSON],
+                  metadata_files: Mapping[str, JSON]
+                  ) -> Iterable[Document]:
+        bundle = api.Bundle(uuid=uuid,
+                            version=version,
+                            manifest=manifest,
+                            metadata_files=metadata_files)
+        project = self._get_project(bundle)
+        samples: MutableMapping[str, Sample] = dict()
+        for file in bundle.files.values():
+            SampleTransformer.get_ancestor_samples(file, samples)
+        for cell_suspension in bundle.biomaterials.values():
+            if not isinstance(cell_suspension, api.CellSuspension):
+                continue
+            visitor = TransformerVisitor()
+            cell_suspension.accept(visitor)
+            cell_suspension.ancestors(visitor)
+            contents = dict(samples=[_sample_dict(s) for s in samples.values()],
```

Comment:
We don't want to associate every cell suspension with every sample. Each CS should only be associated with those samples that are reachable from it via the graph.

Reply:
Ah yes, this makes sense.